commit 10d2baf6578d87ab4ed09be4184dddeae21fec36
Author: Chen Fangzhou <fangzhou@fydeos.io>
Date:   Mon Jan 29 05:32:55 2024 +0000

    add third_party/widevine/cdm/chromeos

diff --git a/third_party/widevine/cdm/chromeos/arm/libwidevinecdm.so b/third_party/widevine/cdm/chromeos/arm/libwidevinecdm.so
new file mode 100644
index 0000000000000..e69de29bb2d1d
diff --git a/third_party/widevine/cdm/chromeos/arm/manifest.json b/third_party/widevine/cdm/chromeos/arm/manifest.json
new file mode 100644
index 0000000000000..04310781aafed
--- /dev/null
+++ b/third_party/widevine/cdm/chromeos/arm/manifest.json
@@ -0,0 +1,38 @@
+{
+  "manifest_version": 2,
+  "update_url": "https://clients2.google.com/service/update2/crx",
+  "name": "WidevineCdm",
+  "description": "Widevine Content Decryption Module",
+  "version": "4.10.2557.0",
+  "minimum_chrome_version": "68.0.3430.0",
+  "x-cdm-module-versions": "4",
+  "x-cdm-interface-versions": "10",
+  "x-cdm-host-versions": "10",
+  "x-cdm-codecs": "vp8,vp09,avc1,av01",
+  "x-cdm-persistent-license-support": true,
+  "x-cdm-supported-encryption-schemes": [
+    "cenc",
+    "cbcs"
+  ],
+  "icons": {
+    "16": "imgs/icon-128x128.png",
+    "128": "imgs/icon-128x128.png"
+  },
+  "platforms": [
+    {
+      "os": "cros",
+      "arch": "x64",
+      "sub_package_path": "_platform_specific/cros_x64/"
+    },
+    {
+      "os": "cros",
+      "arch": "arm",
+      "sub_package_path": "_platform_specific/cros_arm/"
+    },
+    {
+      "os": "cros",
+      "arch": "arm64",
+      "sub_package_path": "_platform_specific/cros_arm64/"
+    }
+  ]
+}
\ No newline at end of file
diff --git a/third_party/widevine/cdm/chromeos/arm64/libwidevinecdm.so b/third_party/widevine/cdm/chromeos/arm64/libwidevinecdm.so
new file mode 100644
index 0000000000000..e69de29bb2d1d
diff --git a/third_party/widevine/cdm/chromeos/arm64/manifest.json b/third_party/widevine/cdm/chromeos/arm64/manifest.json
new file mode 100644
index 0000000000000..04310781aafed
--- /dev/null
+++ b/third_party/widevine/cdm/chromeos/arm64/manifest.json
@@ -0,0 +1,38 @@
+{
+  "manifest_version": 2,
+  "update_url": "https://clients2.google.com/service/update2/crx",
+  "name": "WidevineCdm",
+  "description": "Widevine Content Decryption Module",
+  "version": "4.10.2557.0",
+  "minimum_chrome_version": "68.0.3430.0",
+  "x-cdm-module-versions": "4",
+  "x-cdm-interface-versions": "10",
+  "x-cdm-host-versions": "10",
+  "x-cdm-codecs": "vp8,vp09,avc1,av01",
+  "x-cdm-persistent-license-support": true,
+  "x-cdm-supported-encryption-schemes": [
+    "cenc",
+    "cbcs"
+  ],
+  "icons": {
+    "16": "imgs/icon-128x128.png",
+    "128": "imgs/icon-128x128.png"
+  },
+  "platforms": [
+    {
+      "os": "cros",
+      "arch": "x64",
+      "sub_package_path": "_platform_specific/cros_x64/"
+    },
+    {
+      "os": "cros",
+      "arch": "arm",
+      "sub_package_path": "_platform_specific/cros_arm/"
+    },
+    {
+      "os": "cros",
+      "arch": "arm64",
+      "sub_package_path": "_platform_specific/cros_arm64/"
+    }
+  ]
+}
\ No newline at end of file
diff --git a/third_party/widevine/cdm/chromeos/x64/libwidevinecdm.so b/third_party/widevine/cdm/chromeos/x64/libwidevinecdm.so
new file mode 100755
index 0000000000000..e69de29bb2d1d
diff --git a/third_party/widevine/cdm/chromeos/x64/manifest.json b/third_party/widevine/cdm/chromeos/x64/manifest.json
new file mode 100644
index 0000000000000..04310781aafed
--- /dev/null
+++ b/third_party/widevine/cdm/chromeos/x64/manifest.json
@@ -0,0 +1,38 @@
+{
+  "manifest_version": 2,
+  "update_url": "https://clients2.google.com/service/update2/crx",
+  "name": "WidevineCdm",
+  "description": "Widevine Content Decryption Module",
+  "version": "4.10.2557.0",
+  "minimum_chrome_version": "68.0.3430.0",
+  "x-cdm-module-versions": "4",
+  "x-cdm-interface-versions": "10",
+  "x-cdm-host-versions": "10",
+  "x-cdm-codecs": "vp8,vp09,avc1,av01",
+  "x-cdm-persistent-license-support": true,
+  "x-cdm-supported-encryption-schemes": [
+    "cenc",
+    "cbcs"
+  ],
+  "icons": {
+    "16": "imgs/icon-128x128.png",
+    "128": "imgs/icon-128x128.png"
+  },
+  "platforms": [
+    {
+      "os": "cros",
+      "arch": "x64",
+      "sub_package_path": "_platform_specific/cros_x64/"
+    },
+    {
+      "os": "cros",
+      "arch": "arm",
+      "sub_package_path": "_platform_specific/cros_arm/"
+    },
+    {
+      "os": "cros",
+      "arch": "arm64",
+      "sub_package_path": "_platform_specific/cros_arm64/"
+    }
+  ]
+}
\ No newline at end of file
commit 3dda7ae36cc8ec24337a9f35cb99e1167f22cd91
Author: Chen Fangzhou <fangzhou@fydeos.io>
Date:   Mon Jan 29 06:12:30 2024 +0000

    add widevine_cdm_version.h

diff --git a/third_party/widevine/cdm/chromeos/arm/widevine_cdm_version.h b/third_party/widevine/cdm/chromeos/arm/widevine_cdm_version.h
new file mode 100644
index 0000000000000..57cecf884f5ae
--- /dev/null
+++ b/third_party/widevine/cdm/chromeos/arm/widevine_cdm_version.h
@@ -0,0 +1,16 @@
+// Copyright (c) 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This is a stand-in for a generated file that is available when the
+// Widevine CDM is available.
+
+#ifndef WIDEVINE_CDM_VERSION_H_
+#define WIDEVINE_CDM_VERSION_H_
+
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
+
+#define WIDEVINE_CDM_AVAILABLE
+#define WIDEVINE_CDM_VERSION_STRING "1.4.8.1008"
+
+#endif  // WIDEVINE_CDM_VERSION_H_
diff --git a/third_party/widevine/cdm/chromeos/arm64/widevine_cdm_version.h b/third_party/widevine/cdm/chromeos/arm64/widevine_cdm_version.h
new file mode 100644
index 0000000000000..3c7f1eefd808e
--- /dev/null
+++ b/third_party/widevine/cdm/chromeos/arm64/widevine_cdm_version.h
@@ -0,0 +1,16 @@
+// Copyright (c) 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This is a stand-in for a generated file that is available when the
+// Widevine CDM is available.
+
+#ifndef WIDEVINE_CDM_VERSION_H_
+#define WIDEVINE_CDM_VERSION_H_
+
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
+
+#define WIDEVINE_CDM_AVAILABLE
+#define WIDEVINE_CDM_VERSION_STRING "4.10.2557.0"
+
+#endif  // WIDEVINE_CDM_VERSION_H_
diff --git a/third_party/widevine/cdm/chromeos/x64/widevine_cdm_version.h b/third_party/widevine/cdm/chromeos/x64/widevine_cdm_version.h
new file mode 100644
index 0000000000000..b2ce72862e01b
--- /dev/null
+++ b/third_party/widevine/cdm/chromeos/x64/widevine_cdm_version.h
@@ -0,0 +1,16 @@
+// Copyright (c) 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This is a stand-in for a generated file that is available when the
+// Widevine CDM is available.
+
+#ifndef WIDEVINE_CDM_VERSION_H_
+#define WIDEVINE_CDM_VERSION_H_
+
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
+
+#define WIDEVINE_CDM_AVAILABLE
+#define WIDEVINE_CDM_VERSION_STRING "1.4.9.1008"
+
+#endif  // WIDEVINE_CDM_VERSION_H_
commit e656d5e4a8638d51a39dbdf38758d20d8e112ea2
Author: Chen Fangzhou <fangzhou@fydeos.io>
Date:   Fri Jan 19 01:31:54 2024 +0000

    all prod patches

diff --git a/third_party/widevine/cdm/widevine.gni b/third_party/widevine/cdm/widevine.gni
index 58f073ca562ca..43bde76e96ef9 100644
--- a/third_party/widevine/cdm/widevine.gni
+++ b/third_party/widevine/cdm/widevine.gni
@@ -12,8 +12,9 @@ declare_args() {
   # Google Chrome for Testing and Android. For Fuchsia, see crbug.com/1354347.
   # Can be optionally enabled in Chromium on non-Android platforms. Please see
   # //src/third_party/widevine/LICENSE file for details.
-  enable_widevine = ((is_chrome_branded || is_chrome_for_testing_branded) &&
-                     !is_fuchsia) || is_android
+  # enable_widevine = ((is_chrome_branded || is_chrome_for_testing_branded) &&
+  #                    !is_fuchsia) || is_android
+  enable_widevine = true
 }
 
 # Disable Widevine when running linux-chromeos.
@@ -55,8 +56,9 @@ assert(!enable_media_foundation_widevine_cdm || enable_widevine)
 declare_args() {
   # Widevine CDM is bundled as part of Google Chrome and
   # Google Chrome for Testing builds.
-  bundle_widevine_cdm = enable_library_widevine_cdm &&
-                        (is_chrome_branded || is_chrome_for_testing_branded)
+  # bundle_widevine_cdm = enable_library_widevine_cdm &&
+  #                       (is_chrome_branded || is_chrome_for_testing_branded)
+  bundle_widevine_cdm = true
 
   # Relative root directory to //third_party/widevine/cdm for CDM files.
   # Can be overridden if the CDM files are located in other places.
