From 1c09cb148377c44d407b2ee9795ef85c3b2d8296 Mon Sep 17 00:00:00 2001
From: invalid_git config <unknown@unknown>
Date: Fri, 10 May 2024 06:30:58 +0000
Subject: [PATCH 48/69]

---
 drivers/net/usb/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/usb/Makefile b/drivers/net/usb/Makefile
index 4964f7b32..b5408f8da 100644
--- a/drivers/net/usb/Makefile
+++ b/drivers/net/usb/Makefile
@@ -7,7 +7,7 @@ obj-$(CONFIG_USB_CATC)		+= catc.o
 obj-$(CONFIG_USB_KAWETH)	+= kaweth.o
 obj-$(CONFIG_USB_PEGASUS)	+= pegasus.o
 obj-$(CONFIG_USB_RTL8150)	+= rtl8150.o
-obj-$(CONFIG_USB_RTL8152)	+= r8152.o
+obj-$(CONFIG_USB_RTL8152)	+= r8152/
 obj-$(CONFIG_USB_HSO)		+= hso.o
 obj-$(CONFIG_USB_LAN78XX)	+= lan78xx.o
 obj-$(CONFIG_USB_NET_AX8817X)	+= asix.o
-- 
2.42.0

