Index: kernel-5_15/drivers/bluetooth/btintel.c
===================================================================
--- kernel-5_15.orig/drivers/bluetooth/btintel.c
+++ kernel-5_15/drivers/bluetooth/btintel.c
@@ -2709,6 +2709,7 @@ static int btintel_setup_combined(struct
 			 * Use packet size 24 for the chip
 			 */
 			hdev->wbs_pkt_len = 24;
+      fallthrough;
 		case 0x11:      /* JfP */
 		case 0x12:      /* ThP */
 		case 0x13:      /* HrP */
Index: kernel-5_15/kernel/trace/trace_events.c
===================================================================
--- kernel-5_15.orig/kernel/trace/trace_events.c
+++ kernel-5_15/kernel/trace/trace_events.c
@@ -2036,7 +2036,7 @@ ftrace_set_trace_pidns_read(struct file
 
 	mutex_lock(&event_mutex);
 	if (tr->filtered_ns)
-		snprintf(buf, sizeof(buf), "pid:%lu", tr->filtered_ns->ns.inum);
+		snprintf(buf, sizeof(buf), "pid:%u", tr->filtered_ns->ns.inum);
 	else
 		strcpy(buf, "not set");
 	mutex_unlock(&event_mutex);
