--- a/sound/soc/realtek/common/snd-hifi-realtek.c
+++ b/sound/soc/realtek/common/snd-hifi-realtek.c
@@ -3464,14 +3464,14 @@
 
 	p = pcm->streams[SNDRV_PCM_STREAM_PLAYBACK].substream;
 	for (i = 0; i < playback_substreams; i++) {
-		p->dma_buffer.dev.dev = snd_dma_continuous_data(GFP_KERNEL);
+		p->dma_buffer.dev.dev = NULL;
 		p->dma_buffer.dev.type = SNDRV_DMA_TYPE_DEV_WC_SG;
 		p = p->next;
 	}
 
 	p = pcm->streams[SNDRV_PCM_STREAM_CAPTURE].substream;
 	for (i = 0; i < capture_substreams; i++) {
-		p->dma_buffer.dev.dev = snd_dma_continuous_data(GFP_KERNEL);
+		p->dma_buffer.dev.dev = NULL;
 		p->dma_buffer.dev.type = SNDRV_DMA_TYPE_DEV_WC_SG;
 		p = p->next;
 	}
--- a/sound/soc/realtek/common/snd-realtek.c
+++ b/sound/soc/realtek/common/snd-realtek.c
@@ -3561,14 +3561,14 @@
 
 	p = pcm->streams[SNDRV_PCM_STREAM_PLAYBACK].substream;
 	for (i = 0; i < playback_substreams; i++) {
-		p->dma_buffer.dev.dev = snd_dma_continuous_data(GFP_KERNEL);
+		p->dma_buffer.dev.dev = NULL;
 		p->dma_buffer.dev.type = SNDRV_DMA_TYPE_DEV_WC_SG;
 		p = p->next;
 	}
 
 	p = pcm->streams[SNDRV_PCM_STREAM_CAPTURE].substream;
 	for (i = 0; i < capture_substreams; i++) {
-		p->dma_buffer.dev.dev = snd_dma_continuous_data(GFP_KERNEL);
+		p->dma_buffer.dev.dev = NULL;
 		p->dma_buffer.dev.type = SNDRV_DMA_TYPE_DEV_WC_SG;
 		p = p->next;
 	}
