--- a/Makefile
+++ b/Makefile
@@ -10,6 +10,7 @@
 #EXTRA_CFLAGS += -pedantic
 #EXTRA_CFLAGS += -Wshadow -Wpointer-arith -Wcast-qual -Wstrict-prototypes -Wmissing-prototypes
 
+EXTRA_CFLAGS += -Wno-enum-conversion
 EXTRA_CFLAGS += -Wno-unused-variable
 #EXTRA_CFLAGS += -Wno-unused-value
 EXTRA_CFLAGS += -Wno-unused-label
@@ -252,9 +253,9 @@
 CONFIG_PLATFORM_NV_TK1 = n
 CONFIG_PLATFORM_NV_TK1_UBUNTU = n
 CONFIG_PLATFORM_ARM_SUNxI = n
-CONFIG_PLATFORM_RTK1319 = y
+CONFIG_PLATFORM_RTK1319 = n
 CONFIG_PLATFORM_RTK16XXB = n
-CONFIG_PLATFORM_RTKSTB = n
+CONFIG_PLATFORM_RTKSTB = y
 CONFIG_PLATFORM_AML_S905 = n
 CONFIG_PLATFORM_AML_S905_L3A = n
 CONFIG_PLATFORM_HUANGLONG = n
