From c7322e0fba1c931a719b499673f6be1af17ed163 Mon Sep 17 00:00:00 2001
From: Phinex Hung <phinex@realtek.com>
Date: Tue, 9 Jan 2024 20:31:54 +0800
Subject: [PATCH] abortboot: detect TAB key to use altbootcmd for rescue mode

---
 common/autoboot.c | 24 +++++++++++++++---------
 1 file changed, 15 insertions(+), 9 deletions(-)

diff --git a/common/autoboot.c b/common/autoboot.c
index 5d331991c1..bb0518521a 100644
--- a/common/autoboot.c
+++ b/common/autoboot.c
@@ -40,6 +40,9 @@ DECLARE_GLOBAL_DATA_PTR;
 static int stored_bootdelay;
 static int menukey;
 
+/*TAB keycode */
+#define RESCUE_KEY 9
+
 #if defined(CONFIG_AUTOBOOT_STOP_STR_CRYPT)
 #define AUTOBOOT_STOP_STR_CRYPT	CONFIG_AUTOBOOT_STOP_STR_CRYPT
 #else
@@ -378,7 +381,7 @@ static int abortboot_single_key(int bootdelay)
 	 * Check if key already pressed
 	 */
 	if (tstc()) {	/* we got a key press	*/
-		getchar();	/* consume input	*/
+		menukey = getchar();	/* consume input	*/
 		puts("\b\b\b 0");
 		abort = 1;	/* don't auto boot	*/
 	}
@@ -389,13 +392,11 @@ static int abortboot_single_key(int bootdelay)
 		ts = get_timer(0);
 		do {
 			if (tstc()) {	/* we got a key press	*/
-				int key;
 
 				abort  = 1;	/* don't auto boot	*/
 				bootdelay = 0;	/* no more delay	*/
-				key = getchar();/* consume input	*/
-				if (IS_ENABLED(CONFIG_AUTOBOOT_USE_MENUKEY))
-					menukey = key;
+				menukey = getchar();/* consume input	*/
+
 				break;
 			}
 			udelay(10000);
@@ -405,7 +406,6 @@ static int abortboot_single_key(int bootdelay)
 	}
 
 	putc('\n');
-
 	return abort;
 }
 
@@ -504,10 +504,16 @@ void autoboot_command(const char *s)
 			disable_ctrlc(prev);	/* restore Ctrl-C checking */
 	}
 
-	if (IS_ENABLED(CONFIG_AUTOBOOT_USE_MENUKEY) &&
-	    menukey == AUTOBOOT_MENUKEY) {
-		s = env_get("menucmd");
+	if(menukey) {
+		if (IS_ENABLED(CONFIG_AUTOBOOT_USE_MENUKEY) &&
+		menukey == AUTOBOOT_MENUKEY) 
+			s = env_get("menucmd");
+		else if (menukey == RESCUE_KEY) 
+			s = env_get("altbootcmd");
+		else
+			s = NULL;
 		if (s)
 			run_command_list(s, -1, 0);
 	}
+
 }
