From bc6a751a2a16fb3701277010f241cebe19b8ed2b Mon Sep 17 00:00:00 2001
From: Phinex Hung <phinex@realtek.com>
Date: Wed, 3 Jan 2024 20:08:29 +0800
Subject: [PATCH] u-boot: add u-boot patch and drivers from openwrt 22.03

---
 scripts/Makefile.spl | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/Makefile.spl b/scripts/Makefile.spl
index e450ffd5d5..a4937fa086 100644
--- a/scripts/Makefile.spl
+++ b/scripts/Makefile.spl
@@ -327,9 +327,13 @@ $(obj)/$(SPL_BIN)-dtb.bin: $(obj)/$(SPL_BIN)-nodtb.bin \
 
 $(obj)/$(SPL_BIN).bin: $(obj)/$(SPL_BIN)-dtb.bin FORCE
 	$(call if_changed,copy)
+	$(OBJCOPY) -I binary -O binary --pad-to=$(CONFIG_SPL_PAD_TO) --gap-fill=0xff $@ $@_pad
+	openssl dgst -sha256 -binary $@_pad >> $@_pad
 else
 $(obj)/$(SPL_BIN).bin: $(obj)/$(SPL_BIN)-nodtb.bin FORCE
 	$(call if_changed,copy)
+	$(OBJCOPY) -I binary -O binary --pad-to=$(CONFIG_SPL_PAD_TO) --gap-fill=0xff $@ $@_pad
+	openssl dgst -sha256 -binary $@_pad >> $@_pad
 endif
 
 # Create a file that pads from the end of u-boot-spl-nodtb.bin to bss_end
@@ -559,7 +563,8 @@ targets += $(SPL_OF_LIST_TARGETS)
 
 MKIMAGEFLAGS_$(SPL_BIN).multidtb.fit = -f auto -A $(ARCH) -T firmware -C none -O u-boot \
 	-n "Multi DTB fit image for $(SPL_BIN)" -E \
-	$(patsubst %,-b %,$(SHRUNK_ARCH_DTB))
+	$(patsubst %,-b %,$(SHRUNK_ARCH_DTB)) \
+	$(patsubst %,-b arch/$(ARCH)/dts/%.dtbo,$(subst ",,$(CONFIG_OF_OVERLAY_LIST)))
 
 $(obj)/$(SPL_BIN).multidtb.fit: /dev/null $(SHRUNK_ARCH_DTB) FORCE
 	$(call if_changed,mkimage)
