diff --git a/boot/image-fit.c b/boot/image-fit.c
index 3cc556b727f..f5c912ae4ac 100644
--- a/boot/image-fit.c
+++ b/boot/image-fit.c
@@ -2261,6 +2261,7 @@ int fit_image_load(struct bootm_headers *images, ulong addr,
 
 	comp = IH_COMP_NONE;
 	loadbuf = buf;
+#define BOOT_LOGO_ADDR          0x2F700000
 	/* Kernel images get decompressed later in bootm_load_os(). */
 	if (!fit_image_get_comp(fit, noffset, &comp) &&
 	    comp != IH_COMP_NONE &&
@@ -2272,6 +2273,10 @@ int fit_image_load(struct bootm_headers *images, ulong addr,
 			loadbuf = malloc(max_decomp_len);
 			load = map_to_sysmem(loadbuf);
 		} else {
+#ifndef USE_HOSTCC
+			if((u32)load == BOOT_LOGO_ADDR)
+				max_decomp_len = max(max_decomp_len, SZ_8M);
+#endif
 			loadbuf = map_sysmem(load, max_decomp_len);
 		}
 		if (image_decomp(comp, load, data, image_type,
